src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,151 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        if (hash < 0) {
+            hash *= -1;
+        }
+        int position = hash % table.length;
+        if (table[position] == null) {
+            table[position] = new Node<>(hash, key, value, null);
+        } else {
+            // check of existing nodes
+            Node<K, V> current = table[position];
+            if (current.key == key || current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            while (current.next != null) {
+                if (current.key == key) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            // check the last node
+            if (current.key == key || current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (table == null) {
+            return null;
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        if (hash < 0) {
+            hash *= -1;
+        }
+        int position = hash % table.length;
+        Node<K, V> current = table[position];
+        while (!(current.key == key || current.key != null && current.key.equals(key))) {
+            current = current.next;
+            if (current == null) {
+                break;
+            }
+        }
+        if (current == null) {
+            return null;
+        }
+        return current.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
```

Comment:
The method 'resize' is only used inside the 'MyHashMap' class and should have a 'private' access modifier for encapsulation.